test/controllers/indices_controller_test.rb

```diff
@@ -77,5 +77,9 @@ class IndicesControllerTest < ActionDispatch::IntegrationTest
     login_as(users(:two))
     patch finalize_index_url(@index)
     assert_redirected_to index_path(@index)
+
+    assert_difference('Index.count', 0) do
```

Comment:
This should really be in a separate test for not allowing deletion of a finalized item.  In which case selecting a different index item for just this test would be fine.